tests/integration/ha_tests/application-charm/src/continuous_writes.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""This file is meant to run in the background continuously writing entries to PostgreSQL."""
+import sys
+
+import psycopg2 as psycopg2
+
+
+def continuous_writes(connection_string: str, starting_number: int):
+    """Continuously writes data do PostgreSQL database.
+
+    Args:
+        connection_string: PostgreSQL connection string.
+        starting_number: starting number that is used to write to the database and
+            is continuously incremented after each write to the database.
+    """
+    write_value = starting_number
+
+    try:
+        # Create the table to write records on and also a unique index to prevent duplicate writes.
+        with psycopg2.connect(connection_string) as connection, connection.cursor() as cursor:
+            connection.autocommit = True
+            cursor.execute("CREATE TABLE IF NOT EXISTS continuous_writes(number INTEGER);")
+            cursor.execute(
+                "CREATE UNIQUE INDEX IF NOT EXISTS number ON continuous_writes(number);"
+            )
+    finally:
+        connection.close()
+
+    # Continuously write the record to the database (incrementing it at each iteration).
+    while True:
+        try:
+            with psycopg2.connect(connection_string) as connection, connection.cursor() as cursor:
+                connection.autocommit = True
+                cursor.execute(f"INSERT INTO continuous_writes(number) VALUES({write_value});")
+        except (
+            psycopg2.InterfaceError,
+            psycopg2.OperationalError,
+            psycopg2.errors.ReadOnlySqlTransaction,
+        ):
+            # We should not raise any of those exceptions that can happen when a connection failure
+            # happens, for example, when a primary is being reelected after a failure on the old
+            # primary.
```

Comment:
Is this because the integration test itself should be checking for continuous writes?

Reply:
Exactly. There will be multiple integration tests checking that no writes were missed after different issues happen, so the checks are made on the tests themselves to avoid a lot of different logic here.